From 5c4dbe2e1ac8ff28fff429af5a93139746d38396 Mon Sep 17 00:00:00 2001
From: Wouter-Michiel Vierdag <w-mv@hotmail.com>
Date: Tue, 30 May 2023 21:22:38 +0200
Subject: [PATCH] Update Shapes How-To for new Lasso tool (napari/#5555) (#149)

# Description
This PR adds documentation for the Shapes layer polygon lasso tool that was added
in napari [#5555](https://github.com/napari/napari/pull/5555)

## Type of change
- [x] Fixes or improves existing content

# References
depends on https://github.com/napari/napari/pull/5555

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality

---------

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>
---
 docs/howtos/layers/shapes.md | 46 +++++++++++++++++++++++++++++++-----
 1 file changed, 40 insertions(+), 6 deletions(-)

diff --git a/docs/howtos/layers/shapes.md b/docs/howtos/layers/shapes.md
index fd90350c..d9b108d8 100644
--- a/docs/howtos/layers/shapes.md
+++ b/docs/howtos/layers/shapes.md
@@ -140,12 +140,46 @@ centered on that click.
 `Polygons` can be added using an array of N vertices. Polygons are closed by
 default, and so you don't also need to include the first point at the end of the
 array. The order of the vertices will determine the triangulation of the
-polygon, which can be non-convex, but cannot have holes. The polygon creation
-tool can be selected from the layer control panel or by pressing the `P` key
-when the shapes layer is selected. When adding a polygon each click will add a
-vertex at the clicked location. To finish drawing a polygon you must click the
-`escape` key, which will add a final vertex at the current mouse position and
-complete the polygon. You'll then be able to start adding another one.
+polygon, which can be non-convex, but cannot have holes. For drawing polygons, 
+multiple tools can be used.
+The polygon creation tool can be selected from the layer control panel or by 
+pressing the `P` key when the shapes layer is selected. When adding a polygon 
+each click will add a vertex at the clicked location. To finish drawing a polygon 
+you must click the`escape` key, which will add a final vertex at the current mouse 
+position and complete the polygon. You'll then be able to start adding another one.
+
+`Polygons` can also be created with the polygon lasso creation tool, which can be 
+found in the layer control panel or by pressing `shift+P`. The tool can be used to 
+draw complex `Polygons` with the mouse
+or tablet. When using the mouse, the sequence of events to draw a polygon is as 
+follows:
+
+1. Click mouse (left-click) to begin drawing.
+2. Move mouse — without holding down the mouse button — to draw the polygon.
+3. Click mouse (left-click) or press `escape` to end drawing—the polygon will auto-complete.
+
+The polygon lasso tool can also be used to draw `Polygons` using a tablet. In this case, 
+drawing the polygon is started by touching the tablet screen with the tablet stylus and 
+drawing will continue for as long as the pencil is moved while touching the tablet screen. 
+Note that similar behavior is also available when using a macOS trackpad, using three-finger 
+drag mode.  
+For both modes, vertices are added only if the vertex to be added is at a certain number of screen pixels 
+away from the previous vertex. This value can be adjusted in the settings in napari by going to 
+`File` -> `Preferences` (or `control + shift + P`), then in the menu on the left-clicking on 
+`Experimental` and then adjusting the value of ` Minimum distance threshold of shapes lasso tool`.
+The default is 10 and can be any integer higher than 0 and lower than 50. As with the polygon creation 
+tool drawing the shape can also be finished by pressing the `escape` key.
+
+After finishing drawing a polygon using the polygon lasso tool, an implementation of the [Ramer–Douglas–Peucker 
+algorithm](https://en.wikipedia.org/wiki/Ramer–Douglas–Peucker_algorithm) is applied to reduce the 
+number of vertices that make up the shape, while preserving its contours. The aggressiveness with 
+which the algorithm reduces the number of vertices of the polygon is determined by an `epsilon` parameter, 
+which is a perpendicular distance threshold. Any vertices beyond the threshold will be preserved, so 
+if `epsilon` is set to `0`, no vertices will be removed. With increasing values of `epsilon`, more and 
+more vertices will be removed. The value of `epsilon` can be set in napari by going to 
+`File` -> `Preferences` (or `control + shift + P`), then in the menu on the left-clicking on 
+`Experimental` and then adjusting the value of `RDP epsilon`. The default value is 0.5 and cannot 
+set lower than 0.
 
 `Paths` are like polygons but are not closed or filled in. They can also be
 added using an array of N vertices. The path creation tool can be selected from
